### ISSUE TITLE
[Bug]: BroadcastChannel support broken in version 23

### ISSUE THREAD

**Issue by dfahlander** [electron/electron#37417]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

23.1.1

### What operating system are you using?

macOS

### Operating System Version

Montery 12.6.3

### What arch are you using?

x64

### Last Known Working Electron version

22.3.1. Stops working in 23.0.0.

### Expected Behavior

Setting up a new BroadcastChannel() in two BrowserWindows with `{ contextIsolation: false }` shall work and communicated events to each other.

### Actual Behavior

BrowserChannels do not communicate between BrowserWindows when contextIsolation is false.

### Testcase Gist URL

[Repro](https://gist.github.com/dfahlander/cb9dfccd6e32101843800feade72049c)

Notes to repro:
1. Run it with Electron 22
2. Move the browser window so you see the other browser window behind
3. Click Post message button and see how messages arrive on the other window
4. Run it with Electron 23
5. Repeat step 2 and 3 but now messages aren't propagated to the other window.

### Additional Information

It seems as the BroadcastChannel we see in a window, is the NodeJS version of BroadcastChannel and not the DOM version because it has the `unref()` method that is unique to Node's version. I don't know if this could be the reason why it does not work as in Electron version 22, where BroadcastChannel is absent in the node thread but present and working fine within BrowserWindows.

**Comment by dfahlander** [comment]
Hi! I created a gist and updated the description. In the repro, I found that the issue only occurs when using contextIsolation: false on the browser windows. Still, I believe that should not affect the behavior of BroadcastChannel since it works perfectly well in Electron 22 but not Electron 23. See repro instructions.

**Comment by dfahlander** [comment]
Hi again! Here comes my thoughts regarding the issue, if they could be beneficial somehow. The repro gist I posted verifies that this is an issue when using contextIsolation: false. I assume the issue could also occur in the preload.js script no matter contextIsolation. I suppose the problem is that the DOM version of BroadcastChannel is replaced by the Node version (since they share the exact same name) and that the Node version regards the different BrowserWindows as being the "same client" and therefore not propagate between them (I suppose the node version only propagates between different threads).

An easy solution (I suppose) would be to prioritize the DOM version before the Node version of BroadcastChannel in both preload scripts and ordinary JS in contextIsolation: false.

A perfect solution would probably be that the Node and the browser versions of BroadcastChannel where connected somehow and possible to use as the channel to communicate also between browser and main thread (as its intention would be). If so, maybe replace BroadcastChannel with your own implemention that would propagate and listen to both the DOM and the Node version. BroadcastChannel has a [really tiny API surface](https://developer.mozilla.org/en-US/docs/Web/API/BroadcastChannel) so I suppose the implementation of such a connecting layer would be minimal.

**Comment by MarshallOfSound** [comment]
I believe this is a bug in this PR --> https://github.com/nodejs/node/pull/40532

Specifically the initialization of `BroadcastChannel` should not  be done where it [currently is](https://github.com/nodejs/node/blob/94e1f8f8e139680d1de0531d8aaf7f971cfaa953/lib/internal/bootstrap/node.js#L229-L233) (in bootstrap/node) and rather it should be moved to bootstrap/browser

cc @zcbenz @codebytere

**Comment by serdaryesilmurat** [comment]
I had the same issue for version 23.1.4. Then, it was fixed in 23.2.0.
[37644](https://github.com/electron/electron/issues/37644)

Now I updated electron version to 26.2.4 which reproduces the same issue. BroadcastChannel does not work in this latest version. As mentioned in this issue above, BrodacastChannel seems to be initialized in `bootstrap/node`.

![image](https://github.com/electron/electron/assets/31957973/76056264-7099-4af1-9f13-d0a6f7f2f887)


In working 23.2.0 version, even when I search for `'BrodacastChannel '`, there is no result.

![image](https://github.com/electron/electron/assets/31957973/8737377f-9e65-4aba-aaaa-9ed3305291c4)


Why is BroadcastChannel is initialized in bootstrap/node in 26.2.4? Any ideas?

PS: BrowserWindow is initialized as `contextIsolation: false`.
